core/iwasm/common/wasm_shared_memory.c

```diff
@@ -345,6 +376,22 @@ wasm_runtime_atomic_wait(WASMModuleInstanceCommon *module, void *address,
 
     os_mutex_lock(&wait_info->wait_list_lock);
 
+    /* add waiter address to atomic wait list */
+    AtomicWaitAddressNode *wait_address_node;
```

Comment:
Does the list contain a copy of the list of keys of `wait_map`? If so, why do we need to maintain that list instead of using `wait_map`'s keys?